### ISSUE TITLE
[1.7.10] Fix that allows macOS users to run Optifine on newer JVM versions is not available on 1.7.10

### ISSUE THREAD

**Issue by cocona20xx** [sp614x/optifine#6098]
**Description of Issue**
As specified in closed issue #4824, Optifine crashes due to a NSException when run under a JVM compiled against newer XCode versions. The fix is already available in 1.8.9 and above, but (presumably) was not tested to see if it occurs on 1.7.10's last Optifine version, which it unfortunately does.

**Steps to Reproduce**
1. Install Forge and Optifine under a JVM which was compiled against a new enough version of XCode to cause the exception.
2. Run the game, leading to the crash.

**OptiFine Version**
1.7.10 HD U E7

**Installation Method**
Forge via Multimc. Ran under Java Version Zulu JDK 8 (1.8.0_292, ARM64), and Forge Version 10.13.4.1614

## Other Installed Mods
N/A (no other mods)

## Log Files/Crash Reports
https://paste.ee/p/oUdbw

## F3 Debug Screenshot
N/A, as the game crashes before reaching the the Forge loadscreen.

## Prior Testing
This does not occur under Forge by itself, or Vanilla by itself.

## Additional Information
This fix should be pretty much the exact same as the fix used for 1.8 and above.

**Comment by UntalentedAmateur** [comment]
Doesn't the changelog for 1.17.x versions of Optifine mention its incompatibility with Forge?  Perhaps this is your crash condition?

**Comment by fayer3** [comment]
> 
> 
> Doesn't the changelog for 1.17.x versions of Optifine mention its incompatibility with Forge? Perhaps this is your crash condition?

you are confusing 1.7 with 1.17, this issue is for 1.7

**Comment by UntalentedAmateur** [comment]
> you are confusing 1.7 with 1.17, this issue is for 1.7

Right.  My mistake.  Struck out the original comment, thanks for picking that up.

**Comment by cocona20xx** [comment]
Any news on if/when this can be fixed?

**Comment by farisalamin** [comment]
Is this ever getting fixed? I have to use an old Intel version Coretto java that gives me a third of the frame rate compared to running a native M1 java version, which makes Optifine essentially useless.

**Comment by MoustacheOff** [comment]
> Is this ever getting fixed? I have to use an old Intel version Coretto java that gives me a third of the frame rate compared to running a native M1 java version, which makes Optifine essentially useless.

Same problem for me…

**Comment by r58Playz** [comment]
@farisalamin @MoustacheOff @cocona20xx if you update to the latest version of my m1-multimc-hack fork, this issue is fixed. Commit that fixed it: https://github.com/r58Playz/lwjgl2-m1/commit/2320dbd082fcb0e99d8eaf9fb0ce0f969b8c8ef5

**Comment by cocona20xx** [comment]
nice, ill see if the ManyMC (form of MultiMC for macOS on Apple Silicon) can add the fix as well @r58Playz 
it *would* be nice to have a solution for this on optifine's side, too, since this effects non-apple silicon users as well due to it being related to an oddity in java on macOS

**Comment by MoustacheOff** [comment]
> @farisalamin @MoustacheOff @cocona20xx if you update to the latest version of my m1-multimc-hack fork, this issue is fixed. Commit that fixed it: [r58Playz/lwjgl2-m1@2320dbd](https://github.com/r58Playz/lwjgl2-m1/commit/2320dbd082fcb0e99d8eaf9fb0ce0f969b8c8ef5)

No way,
You're always the best!
